### ISSUE TITLE
Waking up from sleep takes 5 secs

### ISSUE THREAD

**Issue by swatchie-1** [EETagent/T480-OpenCore-Hackintosh#30]
Does anyone have their T480 taking up 5 seconds to wake from sleep?

I have a T440P and it wakes up instantly from sleep.

**Comment by zombillano** [comment]
It wakes up instantly for me, no matter if it wakes up from a long or a short sleep. May I know the exact model of your T480? Also please share your complete setup in terms of display, charger, SSD, WiFi card, etc, etc. The more the better. All this info can be obtained directly from the Lenovo Diagnostic Utility. Please  avoid to share any private info like serial numbers or UUIDs.

I noticed that we have some issues over here that are kinda exclusive to some people and almost impossible to replicate to some others. I want to know if it is setup related. 

Thanks in advance, mate

**Comment by swatchie-1** [comment]
![Screenshot 2020-09-19 at 9 07 03 AM](https://user-images.githubusercontent.com/40285586/93655685-93fab000-fa57-11ea-99e0-c8b1a4a764a5.png)

@zombillano i've been spending days on figuring out the cause of slow sleep and i narrowed it down to the USBPorts.kext. My unit does not have a webcam hence i suspect that may be causing the slow wake. If it's not too troubling, can you take a snippet of your USB Ports from the Hackintool? I'd like to know which port is for the Webcam.

Appreciate it.

**Comment by zombillano** [comment]
Of course, here you have it:

![imagen](https://user-images.githubusercontent.com/34992400/93656773-77785c80-f9f2-11ea-9c70-7c1f6c039ca7.png)

I don't think your problem is USB ports, but please let us know if you solve your problem. BTW, what's the SSD that you are using? Have you tested the r/w speed before and after waking up from sleep?

**Comment by EETagent** [comment]
For me, Big Sur wakes instantly 

> Does anyone have their T480 taking up 5 seconds to wake from sleep?
> 
> I have a T440P and it wakes up instantly from sleep.

For me, Big Sur wakes instantly, like 1-2 seconds.

**Comment by swatchie-1** [comment]
> > Does anyone have their T480 taking up 5 seconds to wake from sleep?
> > I have a T440P and it wakes up instantly from sleep.
> 
> For me, Big Sur wakes instantly, like 1-2 seconds.

Let me try it now. Did you just drag and drop your entire EFI folder in or is there any config changes you had to do for Big Sur?

**Comment by EETagent** [comment]
Just removing CFG lock and framebuffer patching stuff, replacing UHD 620 with 520 and generating new SMBIOS

**Comment by swatchie-1** [comment]
@zombillano Hi, may I know which bios version are you using? I’m using 1.34 modded and plan to upgrade to 1.35.
I’m having issues with the slow wake. In Windows, the brightness keys and all other hot key buttons apart from volume takes 10 secs to react. I’ve reinstalled many times.

@EETagent ive tried Big Sur and the result is the same. Slow wake. 

Thx

**Comment by zombillano** [comment]
> @zombillano Hi, may I know which bios version are you using? I’m using 1.34 modded and plan to upgrade to 1.35.
> I’m having issues with the slow wake. In Windows, the brightness keys and all other hot key buttons apart from volume takes 10 secs to react. I’ve reinstalled many times.

I'm using 1.34 and most likely I won't update since I don't want to loose my BIOS mod. I think you have a hardware problem, based in the fact that you've had several specific issues that neither @EETagent or I were able to reproduce.

Also, I don't have any other suggestion since I don't really now your hardware. I remember you said you have a i5 computer with a KingSpec SSD, but nothing else. May be the SSD, that chinese brand isn't known by their durable products. Remember that macOS is pretty picky about SSD controllers and so.

Also, if you have a MX150 dGPU remember to disable it via config.plist. We just had an issue about it a few hours ago. 

My best suggestion is to run Lenovo's Diagnostics Utility and try to see if it shows any component damaged. Otherwise you may need to talk with a Lenovo representative, since as you mention your issues are OS independent.

**Comment by swatchie-1** [comment]
Thanks Zombillano. I don't mind risking it and flashing it again later. I've updated to 1.35 now and didn't see any improvements. Looks like I will need to hardware flash it again later to enable the menus. My hardware is as follows:

Colour: Black
Intel Core i5-8350U CPU @ 1.70GHz, 4 Core(s)
8GB RAM + 16GB Kingston @ 2400MHz.
256GB NVMe SSD + Kingspec 256GB 
14-inch (1920x1080) Screen
Wireless 8265 / 8275, Wireless
Ethernet Connection (4) I219-LM, Ethernet
Intel, UHD Graphics 620
Intel, Sunrise Point-LP HD Audio
Wi-Fi and Bluetooth
SD Card Reader
Fingerprint Scanner
USB Port (2), USB Type-C Port (1), LAN Port (Yes), HDMI Port (Standard)
Windows 10 Professional
Accessories Included: AC Adapter
Part Number: 20L6S5E200


I'm using the Kingspec for the mac and the 256gb PM981 for the Windows.
I've seen your score on geekbench 5 at 44TDP. Very nice.

Off topic here, when i boot to windows, the hotkey buttons such as brightness, wifi, settings, bluetooth take 15 secs to respond. The lid also take time to register to sleep. Let me keep trying.

**Comment by zombillano** [comment]
> Thanks Zombillano. I don't mind risking it and flashing it again later.

Neither do I, I just don't want to dissasemble my thinkpad if there's no need. Those plastic clips that hold the bottom cover are so easy to break that I prefer to avoid opening it.

> I'm using the Kingspec for the mac and the 256gb PM981 for the Windows.

Are you disabling your PM981 via config.plist file?

> I've seen your score on geekbench 5 at 44TDP. Very nice.
Thanks, I think they might be higher using other OS, but I'm pretty busy at school to make more tests. 

> Off topic here, when i boot to windows, the hotkey buttons such as brightness, wifi, settings, bluetooth take 15 secs to respond. The lid also take time to register to sleep. Let me keep trying.

If SSDs aren't your issue then the only idea that comes to my mind is that you'd have a damaged EC. It can be responsable of most of your issues about sleeping/high CPU usage/Hotkeys taking long to respond, etc, etc. As I said, nobody here were able to reproduce your problems, and I don't see any particular difference  on your setup compared with ours T480, other than the SSDs.

**Comment by EETagent** [comment]
> Neither do I, I just don't want to dissasemble my thinkpad if there's no need. Those plastic clips that hold the bottom cover are so easy to break that I prefer to avoid opening it.

Yep, I actually broke some but it is not recognizable

**Comment by swatchie-1** [comment]
Haven't found a solution to this yet. It could be related to my EC as @zombillano pointed out. Any way to fix my EC?
If you are free, can you take a look at my DSDT.dsl?
[DSDT.dsl.zip](https://github.com/EETagent/T480-OpenCore-Hackintosh/files/5291286/DSDT.dsl.zip)

**Comment by zombillano** [comment]
> Haven't found a solution to this yet. It could be related to my EC as @zombillano pointed out. Any way to fix my EC?
> If you are free, can you take a look at my DSDT.dsl?
> [DSDT.dsl.zip](https://github.com/EETagent/T480-OpenCore-Hackintosh/files/5291286/DSDT.dsl.zip)

You must try with the built in lenovo diagnostics utility, you need to press F12 at boot time to launch it. It is OS independent which is the best way to tell if you have a config problem or a hardware problem. 

I don't know of any way to fix your EC. It may be possible to do if you find the same chip and flash it with a dump of your actual EC, but since you don't know exactly what's failing on your EC then it may be a hit and miss option.

Only real solution would be to contact Lenovo Support and send your T480 to them. You can keep your SSDs if you want, specially the 2242 one, just in case they send you a new/different machine as a replacement of yours. Dissasembling your T480 doesn't break the warranty but any hardware/software mod will, so upgrade your BIOS in order to remove your mod.

About your DSDT: I'm pretty noob when talking about ACPI, so I can't really tell you if there's something wrong with your DSDT.dsl file. I may be able to compare it with mine but I don't think i'll be able to do it soon, you may need to wait 'till the weekend.

**Comment by swatchie-1** [comment]
@zombillano @EETagent Thanks. I've solved it by flashing back to pre-bios unlock v1.34. Now my windows keys are working as per normal, sleep is as per normal. Waking up still takes 5 secs though. Windows wake from sleep takes 2 secs. 

So you were right about the UEFI. I noticed upgrading or downgrading bios does not completely wipe the old settings, hence the windows hotkey problems still persist. Only by flashing it with the ch341a completely to pre-unlock bios will solve it.
